doc/db_schema.md

```diff
@@ -7,7 +7,7 @@ erDiagram
         bytea PrivateKey "The private key of the owner/wallet"
     }
```

Comment:
I am curious as to the choice, of having the private key stored in the database. If someone with malicious intent, gained access to the database, he could potentially derive all private keys, decrypt them, and use them to create Wallet Endpoints? I assume this private key is actually a child private key, derived from the root seed, being the registry. Instead of storing it permanently in the database, when a private key is needed, the system can derive it on-the-fly from the seed, use it, and then discard it immediately, never storing it in a persistent manner. This way, even if an attacker gains access to the system, or database, he cant to find the private keys.

Reply:
The private keys are not stored encrypted as of now. 

The reason why we have to store a private key for each wallet and cannot derive it, is that we intent to allow people to bring their own private-key they could have created from their own seed or a sub-key there of. 

But yes, as of know if a malicious  actor gained access to the database, they would be able to perform actions on behalf of the user, similar if they gained their access token. 

We could possible minimize this by encrypting the private-key and having the decryption key not stored in the database. 

The best solution in the long run is to move secrets as close to the consumers as possible, but most users are not yet ready for digital wallets, and we would still require some soft of ability to act on behalf of the user for automated processes.

And it is always possible for the user to host their own wallet on their own server, and to ensure more isolation that way.